.github/workflows/check_license_header.yaml

```diff
@@ -0,0 +1,87 @@
+name: Check license
+
+on: [pull_request]
+
+jobs:
+  check-licenses:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/javascript-action@v1
+      - uses: actions/checkout@v2
+
+      - name: Check file extensions
+        if: always()          # needed so that it runs even if previous one fails
+        run: |
+          # highlight if in these folders there are files with other extensions
+          find src include test miniapp -type f                                   \
+               ! '('                                                              \
+                       -name '*.cpp'                                              \
+                   -o  -name '*.h'                                                \
+                   -o  -name '*.h.in'                                             \
+                   -o  -name '*.tpp'                                              \
+                   -o  -name '*.cu'                                               \
+                   -o  -name 'CMakeLists.txt'                                     \
+               ')'                                                                \
+            > result-extension.check
+          # Generate an error message for each file with an unkown extension
+          for filepath in `cat result-extension.check`; do                        \
+            echo "::error file=$filepath,line=1::check extension of $filepath";   \
+          done
+
+          test ! -s result-extension.check
+
+      - name: Check source code
+        if: always()          # needed so that it runs even if previous one fails
+        run: |
+          # Compare first lines of each source file with reference in misc/HEADER
+          find src include test miniapp -type f                                   \
+               '('                                                                \
+                       -name '*.cpp'                                              \
+                   -o  -name '*.h'                                                \
+                   -o  -name '*.h.in'                                             \
+                   -o  -name '*.tpp'                                              \
+                   -o  -name '*.cu'                                               \
+               ')'                                                                \
+            | xargs -I{} sh -c                                                    \
+                "grep -sq 'dlaf-no-license-check' {}                              \
```

Comment:
This is where the special tag check is done. It is valid just for source code files.